### PR TITLE
Rename overloaded `to/from_dask_dataframe` API

### PR DESCRIPTION
**Pull request by rjzamora**
Closes https://github.com/dask/dask-expr/issues/905

### CODE REVIEW

**Comment by rjzamora** [comment]
@phofl - I've been finding the overloaded/conflicting meanings of `to/from_dask_dataframe` in `dask.array`/`dask.bag`/`dask-cudf`/etc to be a bit confusing. Hopefully you are still okay with renaming.

You mentioned earlier that we will also need to make changes in dask/dask. Does it make sense to leave the deprecation warning out until we are confident that dask/dask is using the "correct" name?

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -1245,8 +1245,26 @@ def repartition(
...
+        **optimize_kwargs
+            Key-word arguments to pass through to `optimize`.
+        """
+        warnings.warn(
+            "`to_dask_dataframe` is deprecated, please use `to_legacy_collection`.",
+            FutureWarning,
+        )
+        return self.to_legacy_collection(*args, **kwargs)
+
+    def to_legacy_collection(self, optimize: bool = True, **optimize_kwargs) -> _Frame:
```
Comment (phofl) [inline dask_expr/_collection.py:1266]: This is too internal imo, lets just use to_legacy_dataframe

File: dask_expr/_collection.py [hunk 2]
```diff
@@ -4705,8 +4723,25 @@ def from_dict(
...
+        Whether to optimize the graph before conversion.
+    """
+    warnings.warn(
+        "`from_dask_dataframe` is deprecated, please use `from_legacy_collection`.",
+        FutureWarning,
+    )
+    return from_legacy_collection(*args, **kwargs)
+
+
+def from_legacy_collection(ddf: _Frame, optimize: bool = True) -> FrameBase:
```
Comment (phofl) [inline dask_expr/_collection.py:4743]: Same here

**Comment by phofl** [comment]
Yeah leaving the warning out until the next dask/dask release happens would be good

File: dask_expr/_collection.py (hunk 1 above)
Comment (rjzamora) [reply dask_expr/_collection.py]: Yeah, I've always been a bit bothered by the (lack of?) Series/Dataframe distinction, but I agree that collection is probably worse. will use `to_legacy_dataframe`

**Comment by rjzamora** [comment]
Thanks for the review @phofl!  I addressed your comments.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Comment by rjzamora** [comment]
>Can you update the following things in dask/dask:
> - Add the new method to the api reference
> - replace usages of to_dask_dataframe/from_dask_dataframe with the new methods
>
>And create a PR here that adds the warning so that we can merge it before the next dask/dask release happens?

Sounds good. Thanks!

**Pull request merged**